### ISSUE TITLE
Remove hard dependency profiler bundle <=> twig bundle

### ISSUE THREAD

**Issue by Taluu** [symfony/symfony#28177]
**Description**  
You currently cannot use the profiler if you do not have the twig bundle installed. If you do not install the twig-bundle, you'll have an error `The service "web_profiler.controller.profiler" has a dependency on a non-existent service "twig".`, even if you don't need twig in your app (api style application).

The profiler should declare its own twig service, looking into its templates, rather than forcing the dev to have a `templates` directory, and an explicit dependency to `twig-bundle`.

**Comment by nicolas-grekas** [comment]
Would you like to work on it? Someone that cares about this is needed here :)

**Comment by Taluu** [comment]
Sure, I'll give it a try. 

The only thing bothering me is how the twig service should be declared though if there is no bundle (declaring it locally to the profiler)... more specifically for its configuration (cache, ...). But I'll try to figure that out. :}

**Comment by stof** [comment]
Why would we make WebProfilerBundle create its own Twig service instead of using the one from Twigbundle ? This would make it harder for all bundles providing custom collectors, as their profiler templates would have to be registered in the custom Twig service rather than the main one, and any Twig extension they provide to use would also have to be registered in this custom Twig environment. In the end, either we make it harder for all bundles (and that's a BC break), or we make WebProfilerBundle duplicate TwigBundle entirely (copying all the same configuration), which becomes a maintenance nightmare.

Note that WebProfilerBundle defines a composer requirement on TwigBundle precisely for that reason..

**Comment by stof** [comment]
note that WebProfilerBundle is purely about providing the profiler visualization (the part about collecting profiles is configured by FrameworkBundle and HttpKernel). And as this visualization relies on Twig, it seems fine to me to have a dependency to TwigBundle.

**Comment by jvasseur** [comment]
I agree with @stof about how hard it would be to hard do it in a BC and maintainable way. Even if I agree about the general idea of making the profiler use it's own twig instance, the complexity of doing is to high for it to be worth it.

**Comment by Taluu** [comment]
@stof The twig bundle wasn't a required dependency up until c8165db3edcaa4587fe41d16833179fe22d97a7b, which is planned for 4.2. And I agree it's also quite complex. Maybe somehow detecting if the twig service already exists could do the trick ?

**Comment by jvasseur** [comment]
@nicolas-grekas the bundle isn't required before 4.2 (https://github.com/symfony/symfony/blob/4.1/src/Symfony/Bundle/WebProfilerBundle/composer.json) only the bridge.
